contracts/chain-adapters/Scroll_Adapter.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@scroll-tech/contracts/L1/gateways/IL1GatewayRouter.sol";
+import "@scroll-tech/contracts/L1/rollup/IL2GasPriceOracle.sol";
+import "@scroll-tech/contracts/L1/IL1ScrollMessenger.sol";
+import "./interfaces/AdapterInterface.sol";
+
+/**
+ * @title Scroll_Adapter
+ * @notice Adapter contract deployed on L1 alongside the HubPool to facilitate token transfers
+ * and arbitrary message relaying from L1 to L2.
+ */
+contract Scroll_Adapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @notice Used as the gas limit for relaying messages to L2.
+     */
+    uint32 public constant l2GasLimit = 250_000;
```

Comment:
nit: why not set this to immutable and allow it to be passed in on construction?

The advantage is that we don't have to change the contract code if we ever want to deploy a new version with a different gas limit.

Reply:
This was originally set to be immutable, but it was updated here: https://github.com/across-protocol/contracts-v2/pull/381#discussion_r1437736300

These adapters use immutable: 
- Base
- Boba
- Optimism

These use const:
- ZKSync
- Arbitrum

---
Should we be looking to normalize this?

Reply:
We should but not in this PR. `immutable` should only be used if we want to set at construction time, otherwise constant is fine.

Reply:
I'm ok with going immutable if we want to redeploy in the future and change this var without changing contract code. This also makes the unit tests more scalable since we do test specifically for this value

Reply:
Done: https://github.com/across-protocol/contracts-v2/pull/381/commits/300735d8ec1d9b7ee7e127c79f89594ad57b78e8